src/translator/response_builder.cpp

```diff
@@ -9,6 +9,78 @@ void ResponseBuilder::buildQualityScores(Histories &histories, Response &respons
   qualityEstimator_.computeQualityScores(histories, response);
 }
 
+void buildTagAlignment(Response &response) {
+  // Step 0: set up look-up tables
+  ABORT_IF(response.source.numSentences() == 0, "No sentence in source");
+  std::vector<size_t> tokenOffsetTable(response.source.numSentences(), 0);
+  size_t numSourceTokens = response.source.numWords(0);
```

Comment:
What happens if no words?

Reply:
Hi, this still seems to be a problem to me.

```
  ABORT_IF(response.source.numSentences() == 0, "No sentence in source");
```

Empty sentences are allowed, with empty translations (https://github.com/browsermt/bergamot-translator/issues/101#issuecomment-821162120). Think of a setting where an `<b></b>` is sent to our pipeline, is this ABORT ok?

Reply:
How can a single tag with empty body can be passed? What should be return given a single tag with empty body?

Reply:
It's legal.  More or less this is the same as a self-closing tag like `<img blahblah/>`: try to place it based on the neighboring text while preserving hierarchy.

Reply:
What’s the tag’s char index for such scenario? Given the length of empty body is 0.

Reply:
If the entire block of text output is empty then all tags are `[0, 0)`

Reply:
The the length of the sentence is 0.

Reply:
This is C++: all ranges are half-open intervals.  If that's not already the case with tags, it should be.

Reply:
I was thinking at the parser side: a empty string cannot have 0 char index. But it is a special case, so treat it separately: https://github.com/browsermt/bergamot-translator/pull/241/commits/87a990c9a9b76d9d4be07fd8a3845d5801bd6db8